### PR TITLE
Game controller

### PR DESCRIPTION
**Pull request by TL-2015**

### CODE REVIEW

File: src/VisualStudioUI.VSMac/Options/CheckBoxListOptionVSMac.cs [hunk 1]
```diff
@@ -121,9 +196,72 @@ public void OnItemToggled(object sender, EventArgs e)
...
+                    _tableView.SelectRow(selectIndex, false);
+                }
+
+                UpdateButtonsStatus();
+            }
+        }
+
+        private void OnMoveUpClicked(object sender, EventArgs e)
+        {
+            try
```
Comment (BretJohnson) [inline src/VisualStudioUI.VSMac/Options/CheckBoxListOptionVSMac.cs:216]: Why do you have the try/catch here? Is that needed? What might throw an exception that we want to protect against?

File: src/VisualStudioUI/Options/CheckBoxListOption.cs [hunk 2]
```diff
@@ -10,6 +10,11 @@ public class CheckBoxListOption : Option
         public float Width = 354.0f;
         public float Height = 72.0f;
 
+        public string UpToolTip = string.Empty;
+        public string DownToolTip = string.Empty;
+
+        public bool AllowReordering { get; }
+
         public event EventHandler? ListChanged;
```
Comment (BretJohnson) [inline src/VisualStudioUI/Options/CheckBoxListOption.cs:18]: Is the `ListChanged` needed here? I would normally expect the Propery.PropertyChanged notification to be used to be notified of any updates. If we want a notification just when the list order changes (but not when items are checked/unchecked) then why don't we rename it to `ListOrderChanged`.

**Comment by BretJohnson** [comment]
See my two comments above. Otherwise, this looks good to me. Thanks.

File: src/VisualStudioUI.VSMac/Options/CheckBoxListOptionVSMac.cs (hunk 1 above)
Comment (TL-2015) [reply src/VisualStudioUI.VSMac/Options/CheckBoxListOptionVSMac.cs:216]: It is useless here, I have removed .

File: src/VisualStudioUI/Options/CheckBoxListOption.cs (hunk 2 above)
Comment (TL-2015) [reply src/VisualStudioUI/Options/CheckBoxListOption.cs:18]: I check the code, the ListChanged  is used to respond to the checkbox status changed. In this case, it is more usfull than Property. So I save it here.

**Pull request merged**